server/init.go

```diff
@@ -5,18 +5,24 @@ import (
 
 	"github.com/cosmos/cosmos-sdk/crypto/keyring"
 
+	"github.com/cosmos/cosmos-sdk/simapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // GenerateCoinKey returns the address of a public key, along with the secret
 // phrase to recover the private key.
 func GenerateCoinKey(algo keyring.SignatureAlgo) (sdk.AccAddress, string, error) {
 	// generate a private key, with recovery phrase
-	info, secret, err := keyring.NewInMemory().NewMnemonic("name", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, algo)
+	encCfg := simapp.MakeTestEncodingConfig()
```

Comment:
I don't think it's a good idea to create a new Codec here. Maybe let's add a `cdc Codec` argument to this function

Reply:
Sounds good